### PR TITLE
Implement GET route for restaurants in an area

### PR DESCRIPTION
**Pull request by jasonxchen**

### CODE REVIEW

File: openapi_server/apis/restaurants_api.py [hunk 1]
```diff
@@ -38,9 +45,31 @@ async def get_restaurants(
...
-    ...
+    results: List
+    try:
+        async with AsyncClient(base_url="https://maps.googleapis.com") as ac:
+            response = await ac.get(
+                f"/maps/api/place/nearbysearch/json?location={latitude}%2C{longitude}&radius={radius}&type=restaurant&key={os.environ.get('GOOGLE_API_KEY')}"
+            )
+            results = response.json()["results"]
+    except:
+        return Response(status_code=500)
```
Comment (moyara) [inline openapi_server/apis/restaurants_api.py]: missing response body for the error

File: tests/test_restaurants_api.py [hunk 2]
```diff
@@ -1,28 +1,47 @@
...
+
 from fastapi.testclient import TestClient
 
 
 from openapi_server.models.list_restaurants import ListRestaurants  # noqa: F401
 from openapi_server.models.list_reviews import ListReviews  # noqa: F401
+from openapi_server.main import app
 
 
 def test_get_restaurants(client: TestClient):
```
Comment (moyara) [inline tests/test_restaurants_api.py:77]: will it make a real request to googleapi? we would need to mock this google api call.

Comment (jasonxchen) [reply tests/test_restaurants_api.py:77]: It does make a real request. If I mock the api call, would I need to comment out parts of `get_restaurants` in `restaurants_api.py`?

Comment (moyara) [reply tests/test_restaurants_api.py:77]: You should mock AsyncClient in get_restaurants method to return the mocked response. So get_restaurants method works without concerning about the dependency and focus on the method logic.

Comment (jasonxchen) [reply tests/test_restaurants_api.py:77]: So only upon deployment we replace the mock api call with the actual google api call?

Comment (jasonxchen) [reply tests/test_restaurants_api.py:77]: I just discovered that the mock AsyncClient only needs to be defined in the **test file** where the `patch` context manager from the built-in unittest library can handle replacing the responses from AsyncClient at the endpoint being tested with the `patch` decorator. To answer my own question, no code needs to be commented/replaced at any point.

**Review by moyara — APPROVED** [review]

**Pull request merged**